hypershift-operator/main.go

```diff
@@ -196,10 +197,24 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 		}
 		return "", fmt.Errorf("couldn't locate operator container on deployment")
 	}
-	operatorImage, err := lookupOperatorImage(opts.ControlPlaneOperatorImage)
-	if err != nil {
-		return fmt.Errorf("failed to find operator image: %w", err)
+	var operatorImage string
+	if err := wait.PollImmediate(5*time.Second, 30*time.Second, func() (bool, error) {
+		operatorImage, err = lookupOperatorImage(opts.ControlPlaneOperatorImage)
+		if err != nil {
+			return false, err
+		}
```

Comment:
is this because `container.Status` needs some time to be populated?

Reply:
I think so. I can not tell exactly why this happens, I just did observe it happen once and it leads to hard to understand issues down the line (`Deployment invalid: "" is not a valid image`). I thought this would be the best workaround, as it is simple and stupid.